Source/SwiftLintFramework/Extensions/Configuration+Parsing.swift

```diff
@@ -194,14 +199,43 @@ extension Configuration {
                 }
 
             case let .default(disabled: disabledRules, optIn: optInRules):
-                if rule is OptInRule.Type, Set(optInRules).isDisjoint(with: rule.description.allIdentifiers) {
-                    queuedPrintError("\(message), but it is not enabled on " +
-                        "'\(Key.optInRules.rawValue)'.")
-                } else if Set(disabledRules).isSuperset(of: rule.description.allIdentifiers) {
-                    queuedPrintError("\(message), but it is disabled on " +
-                        "'\(Key.disabledRules.rawValue)'.")
+                validateConfiguredRulesAreEnabled(
+                    message: message,
+                    parentConfiguration: parentConfiguration,
+                    disabledRules: disabledRules,
+                    optInRules: optInRules,
+                    rule: rule
+                )
+            }
+        }
+    }
+
+    private static func validateConfiguredRulesAreEnabled(
+        message: String,
+        parentConfiguration: Configuration?,
+        disabledRules: Set<String>,
+        optInRules: Set<String>,
+        rule: Rule.Type
+    ) {
+        if rule is OptInRule.Type {
+            var allOptInRules = optInRules
+            if let parentConfiguration {
+                switch parentConfiguration.rulesMode {
+                case .allEnabled:
+                    return
+                case .only(let parentOnlyRules):
+                    allOptInRules.formUnion(parentOnlyRules)
+                case let .default(disabled: _, optIn: parentOptInRules):
+                    allOptInRules.formUnion(parentOptInRules)
                 }
             }
+            if Set(allOptInRules).isDisjoint(with: rule.description.allIdentifiers) {
+                queuedPrintError("\(message), but it is not enabled on " +
+                                 "'\(Key.optInRules.rawValue)'.")
+            }
+        } else if Set(disabledRules).isSuperset(of: rule.description.allIdentifiers) {
```

Comment:
Rules could also be disabled in the parent.

Reply:
So I think I've taken care of the parent case now, where, for example

```
included:
  - Source
disabled_rules:
  - blanket_disable_command
```

and

```
parent_config: .swiftlint-new.yml

blanket_disable_command:
  always_allow: [file_header, file_length, file_name, file_name_no_space, single_test_class]
```

which I have producing 

`warning: Found a configuration for 'blanket_disable_command' rule, but it is disabled in a parent configuration.`

I'm going to need to take a look at the other cases in more detail, so I'm going to move this back to draft for now. They're basically the inverse of the original bug in a way.

It would be great to have a way to unit test these, as the logic is actually quite complex.